src/lib.rs

```diff
@@ -202,9 +207,45 @@ impl Builder {
         self
     }
 
+    /// Return `Logger` and TLS connection background task.
+    #[cfg(feature = "tokio-rustls")]
+    pub fn connect_tls<T>(
+        self,
+        addr: T,
+        domain_name: String,
+        client_config: std::sync::Arc<tokio_rustls::rustls::ClientConfig>,
+    ) -> Result<(Logger, BackgroundTask), BuilderError>
+    where
+        T: ToSocketAddrs,
+        T: Send + Sync + 'static,
+    {
+        let dnsname = tokio_rustls::webpki::DNSNameRef::try_from_ascii_str(&domain_name)
```

Comment:
There are a lot of `.to_owned()` and `.clone()` in this function, can we reduce it?